Landis+Gyr Support T550

vpathuis opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change adds support for the T550 model by reading MWh-values whenever supplied by the ultraheat-api. 


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
#81464
#85548
#84130
https://github.com/vpathuis/ultraheat/issues/7
#81464
#85012
#86545

- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

vpathuis committed:
Use mwh values from ultraheat api when available

quizilkend commented:
Sorry, to bother with a comment. But I would be very happy if this gets merged. Is there progress lately?

Thanks and best regards!
Samuel

Kugelfang666 commented:
same here, my meter was just changed to a T550 and this integration is not working for me, this update would be much appreciated 👍

vpathuis commented:
Hi @quizilkend @Kugelfang666, thanks for the support. Tbh I 'm out of ideas how to get this PR merged. They are very busy I guess. It's been ready and waiting since oktober last year .... :(  Really sorry about this. If you have any suggestions on how to get it approved, feel free to let me know!

Kugelfang666 commented:
Hmm im not sure about how to trigger the merge. Maybe check the discord server and  ask there?

Kugelfang666 commented:
@vpathuis : this is the feedback I received on discord: 

"...That PR has conflicts, so no core dev is going to look at it while there are conflicts. The author would have to resolve them and push the updated changes. That said, even if they do that, there's no guarantee it will be looked at quickly anyway..."

vpathuis commented:
Hi @Kugelfang666 you're right, the constant changes on the code base require me to keep this PR up to date. I've rebased a great many times by now, which costs a lot of time. I've rebased again and I really hope someone will pick up the review soon.

Kugelfang666 commented:
Sorry to hear that, do i see it right that the only think missing now is someone with writing rights to sign off on this?

vpathuis commented:
Yes. That's right. Also a review of the changes has already been done in #80346, which is the same PR. I'd hoped that moving this to a new PR would help, but so far it hasn't.

Kugelfang666 commented:
in case this is not merged, would you maybe be able to offer this as a custom component through HACKS?

vpathuis commented:
You can always download the files to a folder in config/custom_components. Don't forget to add a (random) version tag to the manifest file.
I'll consider HACKS, but I'd rather focus on getting it approved to core. I'm going to try and split off some parts of the PR to make it less daunting (because it isn't).

vpathuis commented:
I've split of much of the original PR, making it much smaller. I'm resubmitting it as a new PR to prevent confusion.
